### PR TITLE
move types to @suite-common

### PR DESCRIPTION
**Pull request by mroz22**
part of #6509 

I want to do this 374b145485b5432c52623e622c4825a9b9b07831 thats why I needed to do this 3d95f76be7a53b749f1f5c8abcb4dd5e94d4f0f1

invity types, coinjoin types, etc. should they have a full-fledged package?  I was lazy to do so, so I distrubuted all the moved types between wallet-types and suite-types package.

### CODE REVIEW

**Comment by Nodonisko** [comment]
I would vote to have Invite separated as much possible in their own package.

**Comment by mroz22** [comment]
done here 9102f8451464b8ca55fba0f4a5896c98dc0ca87d

maybe also @suite-common/coinjoin-types, since there is aready @suite-common/metadata-types which is similar level of abstraction/organization ?

**Comment by Nodonisko** [comment]
I kind of trying to avoid these type-only packages because packages are more supposed to by per-feature than per-type. Packages like `wallet-types` was made because it was painless and quick for migration to common. 

I would prefer to have packages just `@suite-common/coinjoin` and `@suite-common/invity` where won't be not only types but everything related to this feature.

**Comment by mroz22** [comment]
ok, why not

https://github.com/trezor/trezor-suite/pull/7336/commits/da6eda364a13385fdef742fd00cc4030786590dd creates suite-common/invity package which contains only types at the moment

https://github.com/trezor/trezor-suite/pull/7336/commits/48623f75dd25f2da8cced71466d4d78ec3205b99 creates suite-common/coinjoin package. also only types.

**Comment by mroz22** [comment]
ok, why not

https://github.com/trezor/trezor-suite/pull/7336/commits/da6eda364a13385fdef742fd00cc4030786590dd creates suite-common/invity package which contains only types at the moment

https://github.com/trezor/trezor-suite/pull/7336/commits/48623f75dd25f2da8cced71466d4d78ec3205b99 creates suite-common/coinjoin package. also only types.

File: suite-common/suite-types/package.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     },
     "dependencies": {
         "@reduxjs/toolkit": "1.8.3",
+        "@suite-common/invity": "workspace:*",
```
Comment (Nodonisko) [inline suite-common/suite-types/package.json]: probably not necessary?

File: suite-common/invity/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+    "extends": "../../tsconfig.json",
+    "compilerOptions": {
+        "outDir": "libDev",
+        "rootDir": "../../../"
```
Comment (Nodonisko) [inline suite-common/invity/tsconfig.json]: why this `rootDir`?

**Review by Nodonisko — CHANGES_REQUESTED** [review]

File: suite-common/invity/tsconfig.json (hunk 2 above)
Comment (mroz22) [reply suite-common/invity/tsconfig.json]: that was a miss 3a983db0f235c4cba5a8354683b4cdae0ad024eb

File: suite-common/suite-types/package.json (hunk 1 above)
Comment (mroz22) [reply suite-common/suite-types/package.json]: yes, 3a983db0f235c4cba5a8354683b4cdae0ad024eb

**Review by Nodonisko — APPROVED** [review]

**Pull request merged**